src/media/video/video.controller.ts

```diff
@@ -28,4 +51,106 @@ export class VideoController {
       this.logger.error(err);
     }
   }
+  @UseGuards(AuthGuard)
+  @Post(':videoId/captions')
+  @UseInterceptors(FileInterceptor('file'))
+  async updateCaptions(
+    @UploadedFile() file: Express.Multer.File,
+    @Param('videoId') videoId: string,
+    @Body() body: VideoCaptionDTO,
+  ): Promise<{ code: string; message: string; result: string }> {
+    const { buffer } = file;
+    try {
+      const key = await this.videoService.uploadCaption(videoId, body.key, buffer);
+      return {
+        code: 'SUCCESS',
+        message: 'successfully update attachment options s3 captions',
+        result: key,
+      };
+    } catch (err) {
+      throw new APIException({
+        code: 'E_UPDATE_CAPTIONS',
+        message: err.message,
+        result: null,
+      });
+    }
+  }
+
+  @UseGuards(AuthGuard)
+  @Get(':videoId/captions')
+  async getCaptions(
+    @Param('videoId') videoId: string,
+  ): Promise<{ code: string; message: string; result: Array<string> }> {
+    try {
+      const keys = await this.videoService.getCaptions(videoId);
+      return {
+        code: 'SUCCESS',
+        message: 'successfully get s3 captions',
+        result: keys,
+      };
+    } catch (err) {
+      throw new APIException({
+        code: 'E_GET_CAPTIONS',
+        message: err.message,
+        result: [],
+      });
+    }
+  }
+
+  @UseGuards(AuthGuard)
+  @Delete(':videoId/captions/:filename')
+  async deleteCaptions(
+    @Param('filename') filename: string,
+    @Param('videoId') videoId: string,
+  ): Promise<{ code: string; message: string; result: Array<string> }> {
+    try {
+      const keys = await this.videoService.deleteCaptions(videoId, filename);
+      return {
+        code: 'SUCCESS',
+        message: 'successfully delete s3 captions',
+        result: keys,
+      };
+    } catch (err) {
+      throw new APIException({
+        code: 'E_DELETE_CAPTIONS',
+        message: err.message,
+        result: [],
+      });
+    }
+  }
+
+  @Get(['*.m3u8', '*.mpd'])
+  async getManifestWithSignUrl(@Req() request: Request) {
+    const [key, signature] = decodeURI(request.url).split('videos/')[1].split('?');
+    const manifest = await this.storageService.getFileFromBucketStorage({
+      Key: key,
+    });
+    const signedManifest = await this.videoService.parseManifestWithSignUrl(
+      await manifest.Body.transformToString(),
+      key,
+      signature,
+    );
+    return sanitizeHtml(signedManifest).replace(/&amp;/g, '&');
+  }
+
+  @Get(':videoId/sign')
+  async signUrl(
+    @Headers('Authorization') authorization: string,
+    @Param('videoId') videoId: string,
+  ): Promise<{ message: string; code: string; result: VideoSignResponseDTO }> {
+    if (!isValidUUID(videoId)) {
+      throw new APIException({
+        code: 'E_SIGN_URL',
+        message: `videoId is undefined`,
```

Comment:
isValidUUID is validated uuid, and maybe this error message should be written 'Invalid videoId' because maybe videoId has content but is not a valid uuid